jedi/evaluate/docstrings.py

```diff
@@ -175,15 +342,39 @@ def _execute_array_values(evaluator, array):
 
 @memoize_default(None, evaluator_is_first_arg=True)
 def follow_param(evaluator, param):
-    def eval_docstring(docstring):
+    """
+    Determines a set of potential types for `param` using docstring hints
+
+    :type evaluator: jedi.evaluate.Evaluator
+    :type param: jedi.parser.tree.Param
+
+    :rtype: list
+
+    >>> from jedi.evaluate.docstrings import *  # NOQA
+    >>> from jedi.evaluate.docstrings import _search_param_in_docstr
+    >>> from jedi.evaluate.docstrings import _evaluate_for_statement_string
+    >>> import jedi
+    >>> source = open(jedi.evaluate.docstrings.__file__.replace('.pyc', '.py'), 'r').read()
+    >>> script = jedi.Script(source)
+    >>> evaluator = script._evaluator
+    >>> func = script._get_module().names_dict['follow_param'][0].parent
+    >>> param = func.names_dict['evaluator'][0].parent
+    >>> types = follow_param(evaluator, param)
+    >>> print('types = %r' % (types,))
+    >>> assert len(types) == 1
+    >>> assert types[0].base.name.value == 'Evaluator'
+    """
+    def eval_docstring(docstr):
+        param_str = str(param.name)
         return set(
-            [p for param_str in _search_param_in_docstr(docstring, str(param.name))
-                for p in _evaluate_for_statement_string(evaluator, param_str, module)]
+            [p for string in _search_param_in_docstr(docstr, param_str)
+                for p in _evaluate_for_statement_string(evaluator, string, module)]
         )
     func = param.parent_function
     module = param.get_parent_until()
 
-    types = eval_docstring(func.raw_doc)
+    docstr = func.raw_doc
```

Comment:
raw_doc doesn't exist anymore. once you rebase you will see that. It's better now, because you can just use something like `function_context.py__doc__()`.